src/main/java/com/example/bookshop/exception/DataProcessingException.java

```diff
@@ -0,0 +1,7 @@
+package com.example.bookshop.exception;
+
+public class DataProcessingException extends RuntimeException {
+    public DataProcessingException(String message, Throwable e) {
+        super(message);
```

Comment:
you dont used this parameter
```suggestion
        super(message, e);
```